### PR TITLE
fix: [CDS-34865]: code changes in NG related to the FFs USE_HELM_REPO…

### PR DESCRIPTION
**Pull request by achyuth-harness**
…_FLAGS, HELM_CHART_VERSION_STRICT_MATCH, HELM_VERSION_3_8_0, DELETE_HELM_REPO_CACHE_DIR

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31350)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/HelmValuesFetchTaskNG.java [hunk 1]
```diff
@@ -94,4 +101,34 @@ public LogCallback getLogCallback(CommandUnitsProgress commandUnitsProgress) {
...
+    String helmPath;
+    if (helmVersion == null) {
+      helmVersion = HelmVersion.V2; // Default to V2
+    }
+    switch (helmVersion) {
+      case V3:
+        helmPath = InstallUtils.getHelm3Path();
+        break;
+      case V380:
+        helmPath = InstallUtils.getNewHelm3Path();
```
Comment (shubham-in10se) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/HelmValuesFetchTaskNG.java:115]:
```suggestion
        helmPath = InstallUtils.getHelm380Path();
```

File: 125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java [hunk 2]
```diff
@@ -130,7 +131,7 @@
 public class CDStepHelper {
   @Inject private GitConfigAuthenticationInfoHelper gitConfigAuthenticationInfoHelper;
   @Named("PRIVILEGED") @Inject private SecretManagerClientService secretManagerClientService;
-  @Inject private CDFeatureFlagHelper cdFeatureFlagHelper;
+  @Inject public CDFeatureFlagHelper cdFeatureFlagHelper;
```
Comment (shubham-in10se) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java:134]: why `public` .?

**Review by shubham-in10se — COMMENTED** [review]
add one test where you are also validating the V380 for HelmVersion

**Review by shubham-in10se — CHANGES_REQUESTED** [review]
> Reviewed 9 of 10 files at r1, 1 of 1 files at r2, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31350)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java, line 134 at r1](https://reviewable.io/reviews/harness/harness-core/31350#-MxnHEOK9qFqckj0qqu3-r1-134:-MyBBmLnCiMIfD0lyn4V:b-ey4n0p) ([raw file](https://github.com/harness/harness-core/blob/d8a50460919f1b814ca8338063c14148f7b842ca/125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java#L134)):*
<details><summary><i>Previously, achyuth-harness (Achyuth Prakash) wrote…</i></summary><blockquote>

So that we can inherit this field, i.e. CDStepHelper is extended by K8sStepHelper and NativeHelmStepHelper 
</blockquote></details>

let us make it `protected` in that case.

**Review by achyuth-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31350)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java, line 134 at r1](https://reviewable.io/reviews/harness/harness-core/31350#-MxnHEOK9qFqckj0qqu3-r1-134:-MyBCYon6dZ3lzP5y6yc:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d8a50460919f1b814ca8338063c14148f7b842ca/125-cd-nextgen/src/main/java/io/harness/cdng/CDStepHelper.java#L134)):*
<details><summary><i>Previously, shubham-in10se (Shubham Maheshwari) wrote…</i></summary><blockquote>

let us make it `protected` in that case.
</blockquote></details>

Done.

**Review by shubham-in10se — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31350)* status: :shipit: complete! ✓

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 8 of 10 files at r1, 1 of 1 files at r2, 6 of 6 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31350#-:-MyBFVzfCm8hafVH3mol:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**